### PR TITLE
Create onyphe.py

### PR DESCRIPTION
**Pull request by doomedraven**
as [this](https://github.com/sebdraven/pyonyphe/pull/5) was ignored for almost 2 months, i had to embed the api, so less deps to install :)

### CODE REVIEW

File: plugins/analytics/public/onyphe.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+from core.analytics import OneShotAnalytics
+from core.errors import ObservableValidationError
+from core.observables import Hostname, Ip
+
+default_types = {
+    "Hostname": Hostname,
+    "Ip": Ip,
+}
+
+if sys.version_info.major == 2:
```
Comment (tomchop) [inline plugins/analytics/public/onyphe.py]: Please use six for this: https://pythonhosted.org/six/

File: plugins/analytics/public/onyphe.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    from urlparse import urljoin
+elif sys.version_info.major == 3:
+    from urllib.parse import urljoin
+
+
+"""
```
Comment (tomchop) [inline plugins/analytics/public/onyphe.py]: Please move the module docstring to the top of the file.

File: plugins/analytics/public/onyphe.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return self.value
+
+class Onyphe:
+    """Wrapper around the Onyphe REST
+
+        :param key: The Onyphe API key that can be obtained from your account page (https://www.onyphe.io)
```
Comment (tomchop) [inline plugins/analytics/public/onyphe.py]: Please indicate the type here (`str` I assume?) and leave the instructions as a comment in the docstring.

e.g.

```
"""Wrapper around the Onyphe REST API.

The Onyphe API key that can be obtained from your account page (https://www.onyphe.io)

    :param key: str
    :type key: str
"""

File: plugins/analytics/public/onyphe.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+        if 'page' in kwargs:
+            payload['page'] = kwargs['page']
+
+        self._choose_url(uri)
+
+        data = self._request('get', payload)
+        if data:
+            return data
+
+    def __search(self,query, endpoint, **kwargs):
```
Comment (tomchop) [inline plugins/analytics/public/onyphe.py]: Why double underscore?

File: plugins/analytics/public/onyphe.py [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        return self._prepare_request(quote('/'.join([self.version, 'search',
+                                               endpoint, query])), **kwargs)
+
+    def synscan(self, ip):
+        """Call API Onyphe https://www.onyphe.io/api/v1/synscan/<IP>
+
+            :param ip: IPv4 or IPv6 address
```
Comment (tomchop) [inline plugins/analytics/public/onyphe.py]: Still a `str`, right?

File: plugins/analytics/public/onyphe.py [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+
+        if json_result:
+            result = {}
+            json_string = json.dumps(json_result, sort_keys=True, indent=4, separators=(',', ': ') )
+            results.update(raw=json_string)
+            result['source'] = 'onyphe_query'
+            result['raw'] = json_string
+            observable.add_context(result)
+
+            #extra = default_types[observable.type].get_or_create(value = observable.value)
```
Comment (tomchop) [inline plugins/analytics/public/onyphe.py]: Plz remove these commented-out lines

File: plugins/analytics/public/onyphe.py (hunk 1 above)
Comment (doomedraven) [reply plugins/analytics/public/onyphe.py]: ok done,  was to not have six as deps in original commit :)

File: plugins/analytics/public/onyphe.py (hunk 4 above)
Comment (doomedraven) [reply plugins/analytics/public/onyphe.py]: that is from original https://github.com/sebdraven/pyonyphe/ code

File: plugins/analytics/public/onyphe.py (hunk 5 above)
Comment (doomedraven) [reply plugins/analytics/public/onyphe.py]: yes

Comment (doomedraven) [reply plugins/analytics/public/onyphe.py]: fixed all

**Comment by doomedraven** [comment]
i don't found the rest useful

**Pull request merged**